### PR TITLE
Create StarboardBridge native class with jni_generator

### PR DESCRIPTION
**Pull request by haozheng-cobalt**
b/372559388

This PR configures jni_generator for cobalt_apk_java. 
It creates the corresponding native class StarboardBridge for cobalt/android/apk/app/src/main/java/dev/cobalt/coat/StarboardBridge.java, adhering to Chromium's JNI standards. 
The StarboardBridge native class is implemented as a singleton and is initialized with the same JNIEnv and jobject used when calling the Starboard custom JniEnvExt::Initialize function. 
Additionally, it updates the JNI function calls in starboard/android/shared/application_android.cc to align with Chromium's modern JNI standards.

### CODE REVIEW

**Review by johnxwork — COMMENTED** [review]
Hey Hao, I'm not sure if the PR is ready for review yet since the description has WIP. Ping me when you need reviews.

File: cobalt/android/apk/app/src/main/java/dev/cobalt/coat/StarboardBridge.java [hunk 1]
```diff
@@ -731,7 +747,7 @@ public boolean hasCobaltService(String serviceName) {
     return cobaltServiceFactories.get(serviceName) != null;
   }
 
-  public CobaltService openCobaltService(long nativeService, String serviceName) {
+  public CobaltService openCobaltService(long nativeService, String serviceName) { // ???????????
```
Comment (yell0wd0g) [inline cobalt/android/apk/app/src/main/java/dev/cobalt/coat/StarboardBridge.java:750]: Remove the comment trailer?

File: cobalt/android/apk/app/src/main/java/dev/cobalt/coat/StarboardBridge.java [hunk 2]
```diff
@@ -144,7 +148,18 @@ public StarboardBridge(
...
+    interface Natives {
+        void onStop();
+
+        long currentMonotonicTime();
+
+        // boolean initJNI();
+
+        // long startNativeStarboard();
+
+        // void closeNativeStarboard(long nativeApp);
```
Comment (yell0wd0g) [inline cobalt/android/apk/app/src/main/java/dev/cobalt/coat/StarboardBridge.java:161]: What's the plan for these three, are they commented out on purpose? Would any functionality be broken? Assuming yes and no, I'd suggest a `// TODO(372559388)` explaining the plan.

File: starboard/android/shared/starboard_bridge.h [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+#include <jni.h>
+
+#include "base/android/scoped_java_ref.h"
+#include "base/memory/singleton.h"
+
+namespace starboard {
+namespace android {
+namespace shared {
+
+class StarboardBridge {
```
Comment (yell0wd0g) [inline starboard/android/shared/starboard_bridge.h:27]: Every class needs some documentation, in particular what it does/used for.

File: cobalt/android/apk/app/src/main/java/dev/cobalt/coat/StarboardBridge.java [hunk 4]
```diff
@@ -52,8 +52,12 @@
 import java.util.HashMap;
 import java.util.Locale;
 import java.util.TimeZone;
+import org.chromium.base.annotations.CalledByNative;
+import org.chromium.base.annotations.JNINamespace;
+import org.chromium.base.annotations.NativeMethods;
 
 /** Implementation of the required JNI methods called by the Starboard C++ code. */
+@JNINamespace("starboard::android::shared")
```
Comment (yell0wd0g) [inline cobalt/android/apk/app/src/main/java/dev/cobalt/coat/StarboardBridge.java:60]: Looking at the use of JNINamespace in Chromium [1], I'd say this should be `@JNINamespace(cobalt)`, WDYT?



[1] https://source.chromium.org/search?q=@JNINamespace&ss=chromium

Comment (haozheng-cobalt) [reply cobalt/android/apk/app/src/main/java/dev/cobalt/coat/StarboardBridge.java:60]: This has to match the namespace in which the corresponding JNI native class is defined.

**Pull request merged**